wradlib/io/gdal.py

```diff
@@ -274,6 +274,7 @@ def close(self):
         if self._geo is not None:
             self._geo = None
         if self.ds is not None:
+            shutil.rmtree(self.ds.GetDescription())
```

Comment:
This might need to be the other way round. First get the filename, then set ds to None (effectively closing the dataset) and after that delete the file.